### ISSUE TITLE
[Bug] Listing non-persistent topics does not work correctly

### ISSUE THREAD

**Issue by alpreu** [apache/pulsar#21042]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

Pulsar 3.1.0

### Minimal reproduce step

1. Create a persistent and a non-persistent topic
```
./bin/pulsar-admin topics create-partitioned-topic persistent://public/default/persistent-topic -p 3
./bin/pulsar-admin topics create-partitioned-topic non-persistent://public/default/non-persistent-topic -p 3
```
2. List partitioned topics (works as expected)
```
./bin/pulsar-admin topics list-partitioned-topics public/default
non-persistent://public/default/non-persistent-topic
persistent://public/default/persistent-topic

### What did you expect to see?

3. List topics (docs state it should include both topic domains)
```
./bin/pulsar-admin topics list public/default
persistent://public/default/persistent-topic-partition-2
persistent://public/default/persistent-topic-partition-0
persistent://public/default/persistent-topic-partition-1
non-persistent://public/default/non-persistent-topic-partition-2
non-persistent://public/default/non-persistent-topic-partition-0
non-persistent://public/default/non-persistent-topic-partition-1
```

4. List topics with topic domain flag set to non-persistent
```
./bin/pulsar-admin topics list public/default -td non_persistent
non-persistent://public/default/non-persistent-topic-partition-2
non-persistent://public/default/non-persistent-topic-partition-0
non-persistent://public/default/non-persistent-topic-partition-1
```

### What did you see instead?

3. List topics (bug: only shows the persistent topics)
```
./bin/pulsar-admin topics list public/default
persistent://public/default/persistent-topic-partition-2
persistent://public/default/persistent-topic-partition-0
persistent://public/default/persistent-topic-partition-1
```
4. List topics with topic domain set to non-persistent (bug: does not find the non-persistent topics)
```
./bin/pulsar-admin topics list public/default -td non_persistent
(nothing)

### Anything else?

Retrieving the topic metadata for the non-persistent topic works:
```
./bin/pulsar-admin topics get-partitioned-topic-metadata non-persistent://public/default/on-persistent-topic
{
  "partitions" : 3,
  "deleted" : false
}
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by codelipenghui** [comment]
@alpreu For the non-persistent topic, it can only be listed while clients are connected. You can try to create a producer or consumer to the non-persistent topic and then list the topics again.

> After a while, even the persistent topic is missing from pulsar-admin topics list

If there are no connected producers or consumers. The topic will be deleted automatically by default. You can try to set `brokerDeleteInactiveTopicsEnabled` to false if you don't want the topic to be deleted.

**Comment by alpreu** [comment]
> @alpreu For the non-persistent topic, it can only be listed while clients are connected. You can try to create a producer or consumer to the non-persistent topic and then list the topics again.
> 

Yes, but don't you think the behavior should be consistent? I think it is weird that non-persistent topics can be listed using `list-partitioned-topics` but not using `list-topics`

> > After a while, even the persistent topic is missing from pulsar-admin topics list
> 
> If there are no connected producers or consumers. The topic will be deleted automatically by default. You can try to set `brokerDeleteInactiveTopicsEnabled` to false if you don't want the topic to be deleted.

Should the output of `get-partitioned-topic-metadata` not show `delete: true` in this case?

**Comment by codelipenghui** [comment]
@alpreu 

> @codelipenghui Yes, but don't you think the behavior should be consistent? I think it is weird that non-persistent topics can be listed using list-partitioned-topics but not using list-topics

Yes, I agree with you. From the implementation's perspective, the broker doesn't want to have a znode for a non-persistent topic/partition. But from the user's perspective, it should be consistent.

It should list:

1. Active topics (have producers or consumers)
2. Inactive persistent topics
3. Inactive non-persistent partitions --> the problem
4. Don't list inactive non-persistent topics (non-partitioned topics)

For the last one, the topic has been deleted if no active producers and consumers.
For the 3rd one, we can fix it by generating the partition name by listing the partitioned topic metadata.

> Should the output of get-partitioned-topic-metadata not show delete: true in this case?

For now, you can try to set `brokerDeleteInactivePartitionedTopicMetadataEnabled=true` in broker.conf.
It's also not a good user experience. It should be enabled by default if users enable topic auto-creation and inactive topic deletion. I think maybe it was worried about breaking the existing behavior before
